MSAL/src/MSALErrorConverter.m

```diff
@@ -67,13 +67,21 @@ + (NSError *)MSALErrorFromMSIDError:(NSError *)msidError
         errorCode = [mapValue integerValue];
     }
     
+    NSMutableDictionary *userInfo = nil;
+    if (msidError.userInfo[MSIDHTTPHeadersKey] || msidError.userInfo[MSIDHTTPResponseCodeKey])
+    {
+        userInfo = [NSMutableDictionary new];
+        [userInfo setValue:msidError.userInfo[MSIDHTTPHeadersKey] forKey:MSALHTTPHeadersKey];
+        [userInfo setValue:msidError.userInfo[MSIDHTTPResponseCodeKey] forKey:MSALHTTPResponseCodeKey];
+    }
+    
     return MSALCreateError(domain,
```

Comment:
This is much more complex then need be. You should be able to provide a key mapping dictionary for MSID -> MSAL keys, transform the userInfo dictionary and be done with it.

Reply:
done